test/grdfill/gridfill.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+# Test the -Ag algorithm in grdfill
+# Pull out a small piece of 20m DEM
+gmt grdcut -R0/10/0/10 @earth_relief_20m -Gdata.grd
+# Make a mask for two holes
+gmt grdmath -Rdata.grd 4 6 SDIST 250 GE 0 NAN 7 1.5 SDIST 100 GE 0 NAN ADD 2 DIV = mask.grd
+# Combine to get data with the holes
+gmt grdmath data.grd mask.grd MUL = holes.grd
```

Comment:
@willschlitzer asked about a sample dataset for demonstrating grdfill usage in pygmt earlier today. If it's small enough, I think we should put `holes.grd` with a slightly more descriptive name (e.g., `earth_relief_20m_holes.nc`) in the cache for broader use in examples/tests, rather than generating it each time inside this test. This would be helpful for our efforts to make the gmt examples fully reproducible.

Reply:
I have no problem with this.  I can comment out the commands that created that file we put in the cache. OK?

Reply:
Sounds good

Reply:
So the script is still showing the mask and the raw data so there is no savings in the shell script... It just means we almost repeat all the calls except the grdmath that multiplies data and mask.  But I understand grdmath is a tricky thing in pyGMT.

Reply:
While grdmath is not yet in pygmt, creating holes in grids is easy using other methods. The motivation is more for simplicity for the people reading the examples, so that they don't need to interpret a couple unrelated commands.

Reply:
Perhaps i'll just go back to an earlier version that just shows the holed grid and the result - that avoids the other 4 frames that are just, ehm, fillers.

Reply:
Done